diff --git a/tools/gnome-session-check-accelerated-gles-helper.c b/tools/gnome-session-check-accelerated-gles-helper.c
index 6a4463d..0cf12b2 100644
--- a/tools/gnome-session-check-accelerated-gles-helper.c
+++ b/tools/gnome-session-check-accelerated-gles-helper.c
@@ -183,8 +183,10 @@ main (int argc,
                 goto out;
         }
 
+	renderer = getenv("GLES_RENDERER");
 #ifdef GDK_WINDOWING_X11
-        renderer = get_gles_renderer ();
+	if (!renderer)
+	        renderer = get_gles_renderer ();
 #endif
 
         if (renderer != NULL) {
